server/gamenode/Lobby.ts

```diff
@@ -51,7 +52,55 @@ export class Lobby {
 
         if (this.game) {
             this.sendGameState(this.game);
+        } else {
+            this.sendDeckInfo();
+        }
+    }
+
+    private updateDeck(socket, command, ...args) {
+        const activeUser = this.users.find((u) => u.id === socket.user.username);
+        const userDeck = activeUser.deck.data;
+        const [cardID] = args; // ['cardID']
+
+        // Determine the arrays we are moving between
+        const sourceArray = command === 'Deck' ? userDeck.deckCards : userDeck.sideboard;
+        const targetArray = command === 'Deck' ? userDeck.sideboard : userDeck.deckCards;
+
+        // Find the card in the source array
+        const sourceIndex = sourceArray.findIndex((item) => item.card.id === cardID);
+        if (sourceIndex === -1) {
+            console.log(`Card with UUID ${cardID} not found in source ${command}`);
+            return;
+        }
+
+        // Extract the card entry from the source
+        const sourceEntry = sourceArray[sourceIndex];
+
+        // Decrement the count in the source entry
+        sourceEntry.count -= 1;
+
+        // If count is now zero, remove it from the source array
+        if (sourceEntry.count <= 0) {
+            sourceArray.splice(sourceIndex, 1);
         }
+
+        // Add this card to the target array
+        // Check if the card already exists in the target
+        const targetIndex = targetArray.findIndex((item) => item.card.id === cardID);
+
+        if (targetIndex === -1) {
+            // Card not in target array, add a new entry
+            targetArray.push({
+                count: 1,
+                card: sourceEntry.card || sourceEntry.card // If we removed from source, we need to ensure we still have the card data
```

Comment:
I think there's a bug here - looks like both sides of the `||` are the same thing?